src/test/java/service/FruitShopServiceTest.java

```diff
@@ -0,0 +1,68 @@
+package service;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import service.impl.FruitShopServiceImpl;
+import strategy.OperationHandler;
+import strategy.impl.BalanceOperationImpl;
+import strategy.impl.OperationStrategyServiceImpl;
+import strategy.impl.PurchaseOperationImpl;
+import strategy.impl.ReturnOperationImpl;
+import strategy.impl.SupplyOperationImpl;
+
+public class FruitShopServiceTest {
+    private static final String BANANA = "banana";
+    private static final String APPLE = "apple";
+    private static FruitShopService defaultFruitShopService;
+    private static List<FruitTransaction> defaultParsed;
+    private static Map<String, Integer> defaultExpected;
+
+    @BeforeClass
+    public static void setUp() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationImpl());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationImpl());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationImpl());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationImpl());
+        defaultFruitShopService =
+                new FruitShopServiceImpl(new OperationStrategyServiceImpl(operationHandlerMap));
+        defaultParsed = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, BANANA, 50),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, APPLE, 20),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, BANANA, 10),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, BANANA, 10),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, BANANA, 10)
+        );

```

Comment:
You should test every logic separately from each other. That's mean if you test logic that contains [FruitShopService - you should test only that logic](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop-tests#ensure-that-you-test-your-services-and-their-independently)

Also, you should have tests for every single strategy implantation, cause it's business logic. You shouldn't test it from other classes, cause it won't be a unit test.